di-ipv-core-stub/src/main/java/uk/gov/di/ipv/stub/core/handlers/BasicAuthHandler.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.di.ipv.stub.core.handlers;
+
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+import uk.gov.di.ipv.stub.core.config.CoreStubConfig;
+
+import java.util.Base64;
+
+public class BasicAuthHandler {
+    private static int NUMBER_OF_AUTHENTICATION_FIELDS = 2;
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String AUTHORIZATION_TYPE = "Basic";
+
+    public BasicAuthHandler() {
+        CoreStubConfig.getUserAuth();
+    }
+
+    public Filter authFilter =
+            (Request request, Response response) -> {
+                if (!request.headers().contains(AUTHORIZATION_HEADER) || !authenticated(request)) {
+                    response.header("WWW-Authenticate", AUTHORIZATION_TYPE);
+                    Spark.halt(401, "Not Authenticated");
```

Comment:
We can add some logs here about authentication error details. I thought, So, we can follow to authentication error in logs.

Reply:
We already have some visibility of this with the spark exception handler - this adds exceptions to the response body and renders in the front-end of the stub for debugging purposes. https://github.com/alphagov/di-ipv-stubs/blob/main/di-ipv-core-stub/src/main/java/uk/gov/di/ipv/stub/core/CoreStub.java#L65 

Also think that because these are stubs we can afford to be a bit more relaxed on logging etc.